packages/sonarwhal/docs/contributor-guide/how-to/test-rules.md

```diff
@@ -146,27 +174,69 @@ const serverConfig = {
 }
 ```
 
-## Execute code `before` or `after` collecting the results
+### Throwing an error
 
-In some scenarios you need to execute some code `before` or `after`
-the actual tests (e.g.: if you need to mock a dependency). For those
-cases you can use the `before` and `after` properties of `RuleTest`:
+If you need to force an error in the `connector` when visiting a URL
+you have to make the content `null`. This will force a redirect to
+`test://fail`, thus, causing an exception.
+
+## Testing an external URL
+
+If you need to test an external resource (because you are integrating
+with a third party service) you need to use the property `serverUrl`:
 
 ```ts
 const tests: Array<RuleTest> = [
     {
-        after() {
-            // Code to execute right before calling `connector.close` goes here.
-        }
-        before() {
-            // Code to execute before the creation of the sonarwhal object goes here.
-        },
         name: 'Name of the tests',
-        serverUrl: 'https://example.com',
         reports: [{
             message: 'Message the error will have'
-        }]
+        }],
+        serverUrl: 'https://example.com'
     },
     { ... }
 ];
 ```
+
+## `ruleRunner.testRule()`
```

Comment:
Probably it will be best if you are the one doing it You are the one who wrote that part and knows more about it. It can be in another PR once this is merged :)

Reply:
xDD ok!